src/book.py

```diff
@@ -741,11 +741,168 @@ def _read_bitpanda_pro_trades(self, file_path: Path) -> None:
+                        change = misc.force_decimal(amount_asset)
+                    else:
+                        raise RuntimeError(
+                            f"Unknown asset class {asset_class}: Should be 'Fiat' or "
+                            f"'Cryptocurrency' in row {row} of file {file_path}"
+                        )
+                elif operation in ["Buy", "Sell"]:
+                    if asset_price_currency != config.FIAT:
+                        raise RuntimeError(
+                            "Only Euro is supported as 'Asset market price currency' "
+                            "currency, since price fetching for fiat currencies is not "
+                            "fully implemented yet."
+                        )
+                    change = misc.force_decimal(amount_asset)
+                    change_fiat = misc.force_decimal(amount_fiat)
+                    # Save price in our local database for later.
+                    price = misc.force_decimal(asset_price)
+                    self.price_data.set_price_db(
+                        platform, asset, "EUR", utc_time, price
+                    )
+
+                if change < 0:
+                    raise RuntimeError(
+                        f"Unexpected value for the amount '{change}' of this "
+                        f"{operation} in row {row} of file {file_path}"
+                    )
+
+                self.append_operation(
+                    operation, utc_time, platform, change, asset, row, file_path
+                )
+
+                # add buy / sell operation for fiat currency
+                if operation == "Buy":
+                    self.append_operation(
+                        "Sell", utc_time, platform, change_fiat, "EUR", row, file_path
+                    )
+                elif operation == "Sell":
+                    self.append_operation(
+                        "Buy", utc_time, platform, change_fiat, "EUR", row, file_path
+                    )
+
+                if fee != "-":
+                    self.append_operation(
+                        "Fee",
+                        utc_time,
+                        platform,
+                        misc.force_decimal(fee),
+                        fee_currency,
+                        row,
+                        file_path,
+                    )
+
     def detect_exchange(self, file_path: Path) -> Optional[str]:
         if file_path.suffix == ".csv":
-            with open(file_path, encoding="utf8") as f:
-                reader = csv.reader(f)
-                header = next(reader, None)
+
+            expected_header_row = {
+                "binance": 0,
+                "binance_v2": 0,
+                "coinbase": 0,
+                "coinbase_pro": 0,
+                "kraken_ledgers_old": 0,
+                "kraken_ledgers": 0,
+                "kraken_trades": 0,
+                "bitpanda_pro_trades": 3,
+                "bitpanda": 6,
+            }
```

Comment:
File lines start with 1 not with 0. I would expect bitpanda row 6 to be in line 6 not in 7. I propose to increase the numbers by 1 for easier readability. `(header_row_num + 1)` below has to be adjusted then.